internal/cloud/realm/user_ext_test.go

```diff
@@ -120,7 +126,53 @@ func TestRealmUsers(t *testing.T) {
 				}
 				users, err := client.FindUsers(groupID, testApp.ID, filter)
 				assert.Nil(t, err)
-				assert.Equal(t, []realm.User{email3}, users)
+				assert.Equal(t, 1, len(users))
+				assert.Equal(t, email3.ID, users[0].ID)
+				assert.True(t, users[0].Disabled, fmt.Sprintf("expected %s to be disabled", users[0].Data["email"]))
+				assert.Equal(t, email3.Identities[0].ProviderType, users[0].Identities[0].ProviderType)
+			})
+
+			t.Run("And enable users", func(t *testing.T) {
+				assert.Nil(t, client.EnableUser(groupID, testApp.ID, email1.ID))
+				users1, users1Err := client.FindUsers(groupID, testApp.ID, realm.UserFilter{IDs: []string{email1.ID}})
+				assert.Nil(t, users1Err)
+				email1Enabled := users1[0]
+				assert.False(t, email1Enabled.Disabled, fmt.Sprintf("expected %s to be enabled", email1Enabled.Data["email"]))
+
+				assert.Nil(t, client.EnableUser(groupID, testApp.ID, email3.ID))
+				users3, users3Err := client.FindUsers(groupID, testApp.ID, realm.UserFilter{IDs: []string{email3.ID}})
+				assert.Nil(t, users3Err)
+				email3Enabled := users3[0]
+				assert.False(t, email3Enabled.Disabled, fmt.Sprintf("expected %s to be enabled", email3Enabled.Data["email"]))
+
+				t.Run("And find all enabled users", func(t *testing.T) {
+					users, err := client.FindUsers(groupID, testApp.ID, realm.UserFilter{State: realm.UserStateEnabled})
+					assert.Nil(t, err)
+					assert.Equal(t, 5, len(users))
+
+					emailUsers := make([]realm.User, 0, 3)
+					apiKeyIDs := make([]string, 0, 2)
+					for _, user := range users {
+						assert.Equalf(t, 1, len(user.Identities), "expected user to have only one identity")
+						switch user.Identities[0].ProviderType {
+						case realm.AuthProviderTypeUserPassword:
+							emailUsers = append(emailUsers, user)
+						case realm.AuthProviderTypeAPIKey:
+							apiKeyIDs = append(apiKeyIDs, user.Identities[0].UID)
+						}
+					}
+					assert.Equal(t, []realm.User{email1Enabled, email2, email3Enabled}, emailUsers)
+					assert.Equal(t, []string{apiKey1.ID, apiKey2.ID}, apiKeyIDs)
+				})
+			})
+
+			t.Run("And find enabled user/password users", func(t *testing.T) {
+				filter := realm.UserFilter{State: realm.UserStateEnabled, Providers: []realm.AuthProviderType{realm.AuthProviderTypeUserPassword}, IDs: []string{email1.ID, email2.ID, email3.ID}}
+				users, err := client.FindUsers(groupID, testApp.ID, filter)
```

Comment:
these users are probably back equal to their original states now (after being disabled and enabled)?  would asserting equality on `[]realm.User{email1, email2, email3}` suffice here?